test/mysql-cdc/alter-column-irrelevant.td

```diff
@@ -58,9 +58,10 @@ $ mysql-execute name=mysql
 ALTER TABLE t1 ADD COLUMN f3 INTEGER FIRST;
 INSERT INTO t1 VALUES (3, 3, 3);
 
-# TODO #24975 (ordered column change requires ALTER SOURCE support)
-! SELECT * FROM t1;
-contains:incompatible schema change
+> SELECT * FROM t1;
```

Comment:
@rjobanp This also fails: https://buildkite.com/materialize/test/builds/83271
```
alter-column-irrelevant.td:61:1: error: executing query failed: db error: ERROR: Source error: u115: incompatible schema change: column f1 in table t1 has been altered: ERROR: Source error: u115: incompatible schema change: column f1 in table t1 has been altered
     |
  13 | $ postgres-execute c ... [rest of line truncated for security]
  20 | > CREATE SECRET mysq ... [rest of line truncated for security]
  24 |     PASSWORD SECRET  ... [rest of line truncated for security]
  27 | $ mysql-connect name ... [rest of line truncated for security]
  60 | 
  61 | > SELECT * FROM t1;
     | ^
+++ !!! Error Report
1 errors were encountered during execution
files involved: alter-column-irrelevant.td
```
I'm not sure i understand what this is supposed to test. Should I drop and recreate the source?

Reply:
I believe this should test that the (unchanged column `f1` of the) table `t1` can still be queried after adding columns that change the column order to the table. Note the `FIRST` in `ALTER TABLE t1 ADD COLUMN f3 INTEGER FIRST;`, which specifies the column position.

Reply:
But this is not the case, you can't query f1 anymore I believe.

Reply:
That's true. Hence, we should continue expecting an error and consider creating a new issue if we want to eventually support this.

Reply:
I'm not sure what this test is meant to do. After the columns are reordered, are we missing calls to drop and re-add the subsource? Is that what the comment means by `ordered column change requires ALTER SOURCE support` ? I wouldn't expect that reordering the columns and NOT altering the source would work (which is what the test is doing in this instance)

Reply:
Ok, understood. We should disable this part of the test then. Will this eventually be supported (that is, shall we create an issue) or not?

Reply:
Well if the source is dropped and re-added it should work, but as-is I don't think we can confidently keep ingesting from a table who's columns have been reordered without some sort of user interaction, so I dont think it'll be fixed as-is.